File: src/main/java/no/fint/provider/springer/TestController.java
package no.fint.provider.springer;

import no.fint.event.model.Event;
import no.novari.fint.model.resource.FintLinks;
import no.fint.provider.springer.service.EventHandlerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {

    @Autowired
    private EventHandlerService eventHandlerService;

    @PostMapping
    public Event<FintLinks> handleEvent(@RequestBody Event<FintLinks> input) {
        Event<FintLinks> response = new Event<>(input);
        eventHandlerService.getActionsHandlerMap().get(input.getAction()).forEach(it -> it.accept(response));
        return response;
    }

}

File: src/main/kotlin/no/fint/provider/springer/TestController.kt
package no.fint.provider.springer

import no.fint.event.model.Event
import no.fint.provider.springer.service.EventHandlerService
import no.novari.fint.model.resource.FintLinks
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController(
    private val eventHandlerService: EventHandlerService
) {
    @PostMapping
    fun handleEvent(@RequestBody input: Event<FintLinks>): Event<FintLinks> {
        val response = Event<FintLinks>(input)
        eventHandlerService.actionsHandlerMap()[input.action].forEach { it.accept(response) }
        return response
    }
}
